src/components/FormStep.js

```diff
@@ -233,6 +233,11 @@ const FormStep = ({form, submission, onLogicChecked, onStepSubmitted, onLogout})
   // look up the form step via slug so that we can obtain the submission step
   const formStep = form.steps.find(s => s.slug === slug);
   const currentStepIndex = form.steps.indexOf(formStep);
+
+  // use meaningful document titles
+  const pageTitle = `${config.titlePrefix} - ${formStep.formDefinition}`;
+  useTitle(pageTitle);
```

Comment:
I would wrap this in our own `useTitle` hook to encapsulate the prefix behaviour.

```jsx
// in hooks/useTitle.js

import {useContext} from 'react';
import {useTitle as reactUseTitle} from 'react-use';
import {ConfigContext} from 'Context';

const useTitle = (localTitle) => {
    const {titlePrefix} = useContext(ConfigContext);
    return titlePrefix ? `${titlePrefix} / ${localTitle}` : localTitle;
};

export default useTitle;
```

Usage would then be:

```jsx
import useTitle from 'hooks/useTitle';

const MyComponent = (...) => {
  ...
  useTitle(formStep.formDefinition);
  ...
};
```

1. This ensure that the prefix is _always_ applied without having to concatenate the strings manually
2. If no existing document title was given, the prefix is empty and then we don't include the separator

Reply:
@sergei-maertens  An elegant idea, but "import {useTitle as reactUseTitle} from 'react-use';" in this way is not used anywhere, so where would we set "reactUseTitle" in this structure?